source/dotnet/Library/AdaptiveCards/Rendering/AdaptiveElementRenderers.cs

```diff
@@ -1,16 +1,54 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Reflection;
 using System.Text;
 using System.Threading.Tasks;
 
 namespace AdaptiveCards.Rendering
 {
+    public class AdaptiveActionHandlers
+    {
+        private readonly List<Type> _supportedActions = new List<Type>()
+        {
+            typeof(AdaptiveOpenUrlAction),
+            typeof(AdaptiveSubmitAction),
+            typeof(AdaptiveShowCardAction)
+        };
+
+        /// <summary>
+        /// Adds support for a given action type. Any action in a payload not specifed here will be dropped from the rendered card
+        /// </summary>
+        public void AddSupportedAction<TAction>()
```

Comment:
>AddSupportedAction [](start = 20, length = 18)

Maybe instead of Add and Remove, we should use "EnableAction" and "DisableAction"?

Reply:
Hmm I'm on the fence on this one. Disabled to me feels like it would just gray out the action or something... Any other ideas?

Reply:
I think part of the problem is the name of the class, `AdaptiveActionHandlers`. When I see a method called `Add` on that, I expect to be adding a handler.

Maybe change the class to `AdaptiveSupportedActions`? And then the existing names of `AddSupportedAction` and whatnot make sense. Or we could even simply make them `Add()` at that point

Reply:
Long-term I was hoping to evolve this class to support the inline delegate so you can wire up the action and it's callback at once, but I didn't have time to fully work through that design yet

Reply:
Well the other problem with supporting the callback here is that it's a global callback now, rather than a callback on the instance of the RenderedAdaptiveCard. Essentially it would be what we used to have, where on the Renderer you could wire up the action handlers, which we chose to move away from.

So I'm not sure it'd make sense to support wiring up the callback at this location.

Reply:
What about TryAppendAction or TryAddAction with bool result? My understanding is that an unsupported action would be dropped so we probably want to know that it failed?

Reply:
The callback is given an instance of the RenderedAdaptiveCard which would provide some (but not all) context for every scenario to properly wire up the callback. 

@mrbrennonwilliams this is so the host can "enable" support for new types of actions specifically for them, e.g., "Action.MyThing". By default the Renderer will drop an action that it doesn't understand (which we will provide as a Warning in the RendereredAdaptiveCard object).

Reply:
Thanks for the clarification @matthidinger 

Excuse my questions - just trying to learn more.

I guess allowing host specific actions will introduce a significant level of potential fragmentation in bot functionality?

I can see how this could benefit a host, and in particular developers working against a single host platform - is this feature intended specifically for that use case? I would assume dev's looking for the largest supported platform surface would continue to author against a "lowest common denominator" in terms of functionality.

If it is intended specifically for hosts, should it then be named AddPlatformAction or AddSupportedHostAction?

Cheers

Reply:
Bots are actually only one use-case for Adaptive Cards. They will used in Windows for different scenarios too, like notifications and the new "Timeline" experience, as well as in Outlook for their Email Cards (again coming soon). Custom actions can of course cause fragmentation in the ecosystem, so our guidance to devs will to be cognizant of that fact, but we definitely want the schema elements and actions to be extensible for any type of scenario